syft/presenter/cyclonedx/bom-extension.go

```diff
@@ -35,15 +34,34 @@ type BdComponent struct {
 }
 
 // NewBomDescriptor returns a new BomDescriptor tailored for the current time and "syft" tool details.
-func NewBomDescriptor() *BomDescriptor {
-	versionInfo := version.FromBuild()
-	return &BomDescriptor{
+func NewBomDescriptor(name, version string, srcMetadata source.Metadata) *BomDescriptor {
+	descriptor := BomDescriptor{
 		XMLName:   xml.Name{},
 		Timestamp: time.Now().Format(time.RFC3339),
 		Tool: &BdTool{
 			Vendor:  "anchore",
-			Name:    internal.ApplicationName,
-			Version: versionInfo.Version,
+			Name:    name,
+			Version: version,
 		},
 	}
+
+	switch srcMetadata.Scheme {
+	case source.ImageScheme:
+		descriptor.Component = &BdComponent{
+			Component: Component{
+				Type:    "container",
+				Name:    srcMetadata.ImageMetadata.UserInput,
+				Version: srcMetadata.ImageMetadata.Digest,
```

Comment:
Not a request for change, but the thought just occurred to me — is it worth checking any other CycloneDX tools to see how they handle this exact case?

Reply:
yes